op-node/rollup/driver/state.go

```diff
@@ -457,6 +458,28 @@ func (s *Driver) eventLoop() {
 	}
 }
 
+func (s *Driver) syncStep(ctx context.Context) error {
+	// If we don't need to call FCU to restore unsafeHead using backupUnsafe, keep going b/c
+	// this was a no-op(except correcting invalid state when backupUnsafe is empty but TryBackupUnsafeReorg called).
+	if fcuCalled, err := s.engineController.TryBackupUnsafeReorg(ctx); fcuCalled {
```

Comment:
The `TryBackupUnsafeReorg` and `TryUpdateEngine` calls are also made inside `derivation.Step` (once it gets down to the `EngineQueue`)  still.  They're both gated on having a flag set to say they're necessary so there's no problem, but do we still need the call in both places?
Looks like right now its required for `op-program` since that replaces this state management/driver, but suspect it can be removed once that is all switched over.

Reply:
Yes, the duplicates here are dropped once we get rid of the EngineQueue entirely. And then we get rid of the test/driver duplication once it's wrapped up in a single composed deriver. First `SyncDeriver`, and then when composed of smaller derivers de-duplicated with op-program.